src/lib.rs

```diff
@@ -1805,7 +1806,10 @@ impl Build {
         }
 
         for flag in self.flags_supported.iter() {
-            if self.is_flag_supported(flag).unwrap_or(false) {
+            if self
+                .is_flag_supported_inner(flag, &cmd.path)
+                .unwrap_or(false)
+            {
                 cmd.push_cc_arg((**flag).into());
             }
         }
```

Comment:
A while ago, I was thinking of parallelising this to speed it up.

If you'd like to do this, I recommend creating a new fn is_flags_supported.

You can take a look at compile_objects, if you are interested in doing so

Reply:
That's an interesting idea, but I'd rather do it in a different PR.

One complication: I'd want to make sure that we don't set up any async stuff if everything is already cached...

Reply:
We could iterate over all flags, and then create two vec `supported_flags: Vec<&str>` and `flags_to_check: Vec<(Command, &str)>`.

Doing so would also avoid the duplicate calls I mentioned in the other code review.

Reply:
> That's an interesting idea, but I'd rather do it in a different PR.

That makes sense, I can create a separate issue for it.

Reply:
Opened #1052 for this